### ISSUE TITLE
Compile Error on Debian 12

### ISSUE THREAD

**Issue by autopr3z** [bibendovsky/bstone#499]
Hi! I'm getting a compile error on the main branch. Below is the error it gives me.

[ 80%] Building CXX object src/CMakeFiles/bstone.dir/bstone_saved_game.cpp.o
/home/dan/Git/bstone/src/bstone_saved_game.cpp: In function ‘void bstone::sg_delete_chunk(FourCc, MemoryStream&)’:
/home/dan/Git/bstone/src/bstone_saved_game.cpp:116:14: error: ‘copy_n’ is not a member of ‘std’; did you mean ‘copy’?
  116 |         std::copy_n(
      |              ^~~~~~
      |              copy

I gather 'copy_n' isn't part of std. I am using Debian 12.

Thanks so much for your hard work on this engine! The pace of development is wild, and it's really exciting to see all these improvements.

**Comment by noabody** [comment]
Adding `#include <algorithm>` to `src/bstone_saved_game.h` solves the missing 'copy_n' issue for me on Manjaro Linux but now I have a linker failure:

```
/usr/bin/ld: /tmp/ccc1Hdbj.ltrans10.ltrans.o: in function `LoadTheGame(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
<artificial>:(.text+0x7f72): undefined reference to `bstone::SgKnownFourCc::vers'
/usr/bin/ld: <artificial>:(.text+0x8634): undefined reference to `bstone::SgKnownFourCc::head'
/usr/bin/ld: <artificial>:(.text+0x8682): undefined reference to `bstone::SgKnownFourCc::lvxx'
/usr/bin/ld: /tmp/ccc1Hdbj.ltrans10.ltrans.o: in function `US_ControlPanel(ScanCode)':
<artificial>:(.text+0x90b8): undefined reference to `bstone::SgKnownFourCc::desc'

```

**Comment by bibendovsky** [comment]
Fixed in 8fc3bceaec1b992905b601e139c42da37a3a4f45.

**Comment by autopr3z** [comment]
Works for me! Thanks for the quick response, now I can finally blake the stone.
